src/main/java/reposense/model/FileTypeManager.java

```diff
@@ -79,6 +83,23 @@ public boolean hasSpecifiedFormats() {
         return groups;
     }
 
+    /**
+     * Adds new groups from {@code groupList}. Skips groups that have already been added previously.
+     */
+    public void addGroups(List<FileType> groupList) {
+        for (FileType group : groupList) {
```

Comment:
You could use stream operations to simplify this loop into a single line
for example, `groupList.stream().filter(this::containsGroup).forEach(groups::add)`